api/models/Members.js

```diff
@@ -36,6 +36,25 @@ var hash_password = key => (member, cb) => {
   })
 }
 
+var update_membership = (member, cb) => {
+  Members
+    .findOne(member.id)
+    .exec(function (error, item) {
+      if (error) {
+        return sails.log.error(error)
+      } else if (member.membership_type && (member.membership_type !== item.membership_type)) {
```

Comment:
This was such awesome advice, thanks Rafe, looks a lot better. Is it bad that I spent four pomodoros trying to make this pointfree? `curry`, `flip`, `__` and loads more got a run out before I realised it was as easy as giving adding another callback. Was a really fun problem.